### PR TITLE
[Autoscaler] Make `setup_commands` in defaults.yaml Docker Compatible

### LINKED ISSUE #14802: [autoscaler] Improve default setup commands for cluster configs with Docker
### What is the problem?

The way default setup_commands are set currently for configs that use docker is confusing.
See
https://discuss.ray.io/t/cannot-start-ray-runtime-on-gcp/1011/10
https://github.com/ray-project/ray/pull/14750#issuecomment-802696221

The default setup commands are geared towards launching Ray clusters without docker, but these days we generally recommend running with Docker.

In case a user provides a docker config but not setup commands, we should fill in sensible defaults, maybe based on the current contents of the example-full yamls.
(Note -- I don't mean we need to literally use the example-full configs programmatically as we do with defaults.yaml.) 

*Ray version and other system information (Python version, TensorFlow version, OS):*

### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [ ] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

### PR DESCRIPTION
**Pull request by ijrsvt**
## Why are these changes needed?
* Some VM-specific `setup_commands` break things inside docker. 
* This PR **preserves the behavior** on VMs, but prevents their execution in Docker.

## Related issue number

* Closes: #14802

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] GCP `defaults.yaml` w/ `min_workers=1`
   - [x] GCP `example-full.yaml` w/ `min_workers=1`
   - [x] AWS `defaults.yaml` w/ `min_workers=1`
   - [x] AWS `example-full.yaml` w/ `min_workers=1`

### CODE REVIEW

File: python/ray/autoscaler/gcp/defaults.yaml [hunk 1]
```diff
@@ -141,17 +141,8 @@ setup_commands:
...
     # below with a git checkout <your_sha> (and possibly a recompile).
     # - echo 'export PATH="$HOME/anaconda3/envs/tensorflow_p36/bin:$PATH"' >> ~/.bashrc
-
-    # Install MiniConda.
-    - >-
-      wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/anaconda3.sh
-      || true
-      && bash ~/anaconda3.sh -b -p ~/anaconda3 || true
-      && rm ~/anaconda3.sh
-      && echo 'export PATH="$HOME/anaconda3/bin:$PATH"' >> ~/.profile
```
Comment (ijrsvt) [inline python/ray/autoscaler/gcp/defaults.yaml:151]: Conda is installed on the default image already

Comment (ijrsvt) [reply python/ray/autoscaler/gcp/defaults.yaml:151]:
```
    Running `which conda`
      Full command is `ssh -tt -i /home/ubuntu/.ssh/ray-autoscaler_gcp_us-west1_anyscale-dev_ubuntu_0.pem -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ConnectTimeout=120s ubuntu@35.247.87.210 bash --login -c -i 'true && source ~/.bashrc && export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (which conda)'`
Warning: Permanently added '35.247.87.210' (ECDSA) to the list of known hosts.
/opt/conda/bin/conda
```

File: python/ray/autoscaler/azure/defaults.yaml [hunk 2]
```diff
@@ -130,7 +130,7 @@ setup_commands:
     - echo 'eval "$(conda shell.bash hook)"' >> ~/.bashrc
```
Comment (ijrsvt) [inline python/ray/autoscaler/azure/defaults.yaml:130]: @gramhagen Is this still necessary with the Azure Data Science VM? Shouldn't Conda be preconfigured?

File: python/ray/autoscaler/azure/defaults.yaml [hunk 3]
```diff
@@ -130,7 +130,7 @@ setup_commands:
...
     # - echo 'conda activate py37_pytorch' >> ~/.bashrc
     - echo 'conda activate py37_tensorflow' >> ~/.bashrc
```
Comment (ijrsvt) [inline python/ray/autoscaler/azure/defaults.yaml:132]: I was also wondering if this is a valid conda environment?

File: python/ray/autoscaler/azure/defaults.yaml (hunk 2 above)
Comment (gramhagen) [reply python/ray/autoscaler/azure/defaults.yaml:130]: yes for some reason the conda env isn't initialized when you ssh in (at least last I tried) and this command is needed to enable that.

File: python/ray/autoscaler/azure/defaults.yaml (hunk 3 above)
Comment (gramhagen) [reply python/ray/autoscaler/azure/defaults.yaml:132]: yes, both of those envs are valid in the dsvm

**Comment by DmitriGekhtman** [comment]
Awesome thanks for fixing these!
 -- would just test it out the AWS and GCP configs manually if you haven't already.

Specifically, we want to make sure that the following works for AWS and GCP
defaults.yaml with min_workers modified to 1
example-full.yaml with min_workers modified to 1 and setup commands deleted.

File: python/ray/autoscaler/azure/defaults.yaml (hunk 3 above)
Comment (ijrsvt) [inline python/ray/autoscaler/azure/defaults.yaml:132]:
```suggestion
    - conda activate py37_tensorflow &> /dev/null && echo 'conda activate py37_tensorflow' >> ~/.bashrc
```

File: python/ray/autoscaler/azure/defaults.yaml [hunk 4]
```diff
@@ -129,8 +129,8 @@ setup_commands:
     # below with a git checkout <your_sha> (and possibly a recompile).
     - echo 'eval "$(conda shell.bash hook)"' >> ~/.bashrc
     # - echo 'conda activate py37_pytorch' >> ~/.bashrc
-    - echo 'conda activate py37_tensorflow' >> ~/.bashrc
-    - pip install -U https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-2.0.0.dev0-cp37-cp37m-manylinux2014_x86_64.whl
+    - conda activate py37_tensorflow &> /dev/null && echo 'conda activate py37_tensorflow' >> ~/.bashrc
```
Comment (ijrsvt) [inline python/ray/autoscaler/azure/defaults.yaml:132]:
```suggestion
    - (conda activate py37_tensorflow &> /dev/null && echo 'conda activate py37_tensorflow' >> ~/.bashrc) || true
```

File: python/ray/autoscaler/azure/defaults.yaml [hunk 5]
```diff
@@ -129,8 +129,8 @@ setup_commands:
     # below with a git checkout <your_sha> (and possibly a recompile).
     - echo 'eval "$(conda shell.bash hook)"' >> ~/.bashrc
```
Comment (ijrsvt) [inline python/ray/autoscaler/azure/defaults.yaml:130]:
```suggestion
    - (which conda && echo 'eval "$(conda shell.bash hook)"' >> ~/.bashrc) || true
```

**Comment by AmeerHajAli** [comment]
@DmitriGekhtman , can you approve/test this PR ?

**Review by gramhagen — APPROVED** [review]
azure part looks good

**Review by DmitriGekhtman — APPROVED** [review]

**Comment by DmitriGekhtman** [comment]
gcp/aws looks good!

**Pull request merged**